amqp_influxdb/__init__.py

```diff
@@ -39,10 +42,31 @@ def __init__(self,
                  exchange,
                  routing_key,
                  message_processor,
-                 connection_parameters=None):
+                 connection_parameters={}):
+        """
+            AMQPTopicConsumer initialisation expects a connection_parameters
+            dict as provided by the __main__ of amqp_influx.
+        """
         self.message_processor = message_processor
 
-        connection_parameters = connection_parameters or {}
+        credentials = connection_parameters.get('credentials', {})
+        credentials_object = pika.credentials.PlainCredentials(
+            username=credentials.get('username', 'guest'),
```

Comment:
IIRC I did this to avoid test failures where username&password weren't supplied via the __main__ (e.g. by using this class directly). This also maintains the 'always default to guest/guest' that we discussed before.

Happy to remove it anyway if you still prefer (assuming it doesn't cause tests to fail)?